src/storage/UpdateEdgeProcessor.cpp

```diff
@@ -0,0 +1,373 @@
+   }
+   return data;
+}
+
+
+cpp2::ErrorCode UpdateEdgeProcessor::checkAndBuildContexts(
+        const cpp2::UpdateEdgeRequest& req) {
+    auto retCode = QueryBaseProcessor<cpp2::UpdateEdgeRequest,
+         cpp2::UpdateResponse>::checkAndBuildContexts(req);
+    if (retCode != cpp2::ErrorCode::SUCCEEDED) {
+        return retCode;
+    }
+
+    // build context of the update items
+    if (this->expCtx_ == nullptr) {
+        this->expCtx_ = std::make_unique<ExpressionContext>();
+    }
+    for (auto& item : req.get_update_items()) {
+        std::string edgeName = item.get_name();
+        std::string edgeProp = item.get_field();
+        auto* edgePropExp = new AliasPropertyExpression(new std::string(""), &edgeName, &edgeProp);
+        edgePropExp->setContext(this->expCtx_.get());
+        auto status = edgePropExp->prepare();
+        if (!status.ok() || !this->checkExp(edgePropExp)) {
+            return cpp2::ErrorCode::E_INVALID_UPDATER;
+        }
+
+        StatusOr<std::unique_ptr<Expression>> exp = Expression::decode(item.get_value());
+        if (!exp.ok()) {
+            return cpp2::ErrorCode::E_INVALID_UPDATER;
+        }
+        auto vexp = std::move(exp).value();
+        vexp->setContext(this->expCtx_.get());
+        status = vexp->prepare();
+        if (!status.ok() || !this->checkExp(vexp.get())) {
+            return cpp2::ErrorCode::E_INVALID_UPDATER;
+        }
+    }
+
+    if (expCtx_->hasDstTagProp()) {
+        VLOG(3) << "ERROR: update edge[at storage level] should not contain "
+                << "any DestPropertyExpression expression!";
+        return cpp2::ErrorCode::E_INVALID_UPDATER;
+    }
+    return cpp2::ErrorCode::SUCCEEDED;
+}
+
+
+void UpdateEdgeProcessor::process(const cpp2::UpdateEdgeRequest& req) {
+    VLOG(3) << "Receive UpdateEdgeRequest...";
+    spaceId_ = req.get_space_id();
+    returnColumnsNum_ = req.get_return_columns().size();
+    VLOG(3) << "Receive request, spaceId " << spaceId_ << ", return cols " << returnColumnsNum_;
+    insertable_ = req.get_insertable();
+    auto partId = req.get_part_id();
+    auto retCode = checkAndBuildContexts(req);
+    if (retCode != cpp2::ErrorCode::SUCCEEDED) {
+        LOG(ERROR) << "Failure build contexts!";
+        this->pushResultCode(retCode, partId);
+        this->onFinished();
+        return;
+    }
+    updateItems_ = std::move(req.get_update_items());
+
+    CHECK_NOTNULL(kvstore_);
+    auto edgeKey = req.get_edge_key();
+    VLOG(3) << "Process spaceId: " << this->spaceId_
+            << ", partId:  " << partId
+            << ", src: " << edgeKey.get_src()
+            << ", edge_type: " << edgeKey.get_edge_type()
+            << ", dst: " << edgeKey.get_dst()
+            << ", ranking: " << edgeKey.get_ranking();
+    kvstore::ResultCode ret = kvstore::ResultCode::SUCCEEDED;
+    this->kvstore_->asyncAtomicOp(spaceId_, partId,
+        [&, this] () -> std::string {
+            ret = this->processEdge(partId, edgeKey);
```

Comment:
Use another name for "processEdge/processData",   i don't know what are you doing inside. 

And for update,   the code structure should look like 

```
auto edgeValue = getEdge()
if (checkFilter()) {
      updateAndWriteBackEdge()
      return true;
} else {
      return false;
}
```

Reply:
ok, I will update it as you suggested.

Reply:
done!